portality/tasks/ingestarticles.py

```diff
@@ -365,29 +387,38 @@ def _file_upload(cls, username, f, schema, previous):
         try:
             actual_schema = None
             with open(xml) as handle:
-                actual_schema = article.check_schema(handle, schema)
-        except:
-            # file is a dud, so remove it
+                actual_schema, xwalk, doc = article.check_schema(handle, schema)
+        except article.IngestException as e:
+            record.failed(e.message, e.inner_message)
             try:
-                os.remove(xml)
+                file_failed(xml)
             except:
                 pass
-
-            # if we're unable to validate the file, we should record this as
-            # a file error.
-            record.failed("Unable to parse file")
             record.save()
             previous.insert(0, record)
-            raise BackgroundException("Failed to parse file - it is invalid XML; please fix it before attempting to upload again.")
+            raise BackgroundException("Failed to upload file: " + e.message + "; " + str(e.inner_message))
+        except Exception as e:
+            record.failed("File system error when reading file")
+            try:
+                file_failed(xml)
+            except:
+                pass
+            record.save()
+            previous.insert(0, record)
+            raise BackgroundException("Failed to upload file - please contact an administrator")
 
-        if actual_schema:
+        if actual_schema is not None:
             record.validated(actual_schema)
             record.save()
             previous.insert(0, record)
         else:
+            # This code should never get executed
             record.failed("File could not be validated against a known schema")
             record.save()
-            os.remove(xml)
+            try:
+                file_failed(xml)
+            except:
+                pass
```

Comment:
At least log? Swallowing silently can be really difficult to debug, in that it starts with "put logging in place".